tests/framework/extensions/charts/ranchercisbenchmark.go

```diff
@@ -0,0 +1,198 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/api/steve/catalog/types"
+	catalogv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/defaults"
+	kubenamespaces "github.com/rancher/rancher/tests/framework/extensions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	// Namespace that rancher CIS Benchmark chart is installed in
+	RancherCisBenchmarkNamespace = "cis-operator-system"
+	// Name of the rancher CIS Benchmark chart
+	RancherCisBenchmarkName = "rancher-cis-benchmark"
+	// Name of rancher CIS Benchmark crd chart
+	RancherCisBenchmarkCRDName = "rancher-cis-benchmark-crd"
+)
+
+// InstallRancher CIS BenchmarkChart is a helper function that installs the rancher-CIS Benchmark chart.
+func InstallRancherCisBenchmarkChart(client *rancher.Client, installOptions *InstallOptions) error {
+	serverSetting, err := client.Management.Setting.ByID(serverURLSettingID)
+	if err != nil {
+		return err
+	}
+
+	registrySetting, err := client.Management.Setting.ByID(defaultRegistrySettingID)
+	if err != nil {
+		return err
+	}
+
+	cisbenchmarkChartInstallActionPayload := &payloadOpts{
+		InstallOptions:  *installOptions,
+		Name:            RancherCisBenchmarkName,
+		Namespace:       RancherCisBenchmarkNamespace,
+		Host:            serverSetting.Value,
+		DefaultRegistry: registrySetting.Value,
+	}
+
+	chartInstallAction := newCisBenchmarkChartInstallAction(cisbenchmarkChartInstallActionPayload)
+
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	// Cleanup registration
+	client.Session.RegisterCleanupFunc(func() error {
+		// UninstallAction for when uninstalling the rancher-CIS Benchmark chart
+		defaultChartUninstallAction := newChartUninstallAction()
+
+		err = catalogClient.UninstallChart(RancherCisBenchmarkName, RancherCisBenchmarkNamespace, defaultChartUninstallAction)
```

Comment:
Can we add this uninstall section as a separate function and then call that function here. It would be easier to have it separately when we want to run tests.

Reply:
I think this was based on the comments [here](https://github.com/rancher/rancher/pull/41981#discussion_r1247898259).

Do you still want me to move the uninstall section into new one?

Reply:
Could you please also elaborate [this comment](https://github.com/rancher/rancher/pull/41981#discussion_r1248093219) as well?

Reply:
Yes Vivek please. And Just name it `UninstallRancherCisBenchmarkChart`

Reply:
Okay